Compiler failure - abstract function inlining

I triggered this error while trying something out on try.haxe - https://try.haxe.org/#3C5532f4
Removing either of the `inline`'s makes it compile successfully.

```Haxe
class Test {
	static function main() {
		trace(Abs.use(fn)(101));
	}

	inline static function fn(v:Int):Int
		return v;
}

@:callable abstract Abs<T>(T->Int) from T->Int {
	inline public static function use<T>(fn:T->Int):Abs<T> {
		return fn;
	}
}
```

```
Compiler failure
Please submit an issue at https://github.com/HaxeFoundation/haxe/issues/new
Attach the following information:
Haxe: 4.3.0; OS type: unix;
File "src/optimization/optimizer.ml", line 356, characters 74-81
Called from file "src/optimization/optimizer.ml", line 356, characters 67-81
Called from file "list.ml", line 92, characters 20-23
Called from file "src/core/texpr.ml", line 136, characters 30-43
Called from file "src/optimization/optimizer.ml", line 340, characters 9-42
Called from file "list.ml", line 92, characters 20-23
Called from file "src/core/texpr.ml", line 131, characters 26-41
Called from file "src/optimization/optimizer.ml", line 340, characters 9-42
Called from file "src/core/texpr.ml", line 140, characters 49-61
Called from file "src/optimization/optimizer.ml", line 340, characters 9-42
Called from file "src/core/tUnification.ml", line 458, characters 10-15
Called from file "list.ml", line 121, characters 24-34
Called from file "src/core/tUnification.ml", line 458, characters 10-15
Called from file "src/filters/filtersCommon.ml", line 75, characters 23-71
Called from file "list.ml", line 110, characters 12-15
Called from file "src/filters/filtersCommon.ml", line 81, characters 2-46
Called from file "list.ml", line 110, characters 12-15
Called from file "src/filters/filters.ml", line 981, characters 1-96
Called from file "src/compiler/compiler.ml", line 285, characters 1-38
Called from file "src/compiler/compiler.ml", line 315, characters 2-17
Called from file "src/compiler/compiler.ml", line 334, characters 1-5
[ERROR] (unknown position)

   | File "src/core/globals.ml", line 156, characters 1-7: Assertion failed
```

--- a/src/optimization/optimizer.ml
+++ b/src/optimization/optimizer.ml
@@ -358,7 +358,7 @@ let rec reduce_loop ctx e =
 				begin match cf.cf_expr with
 				| Some {eexpr = TFunction tf} ->
 					let config = inline_config (Some cl) cf el e.etype in
-					let rt = (match follow e1.etype with TFun (_,rt) -> rt | _ -> die "" __LOC__) in
+					let rt = (match Abstract.follow_with_abstracts e1.etype with TFun (_,rt) -> rt | _ -> die "" __LOC__) in
 					let inl = (try type_inline ctx cf tf ef el rt config e.epos false with Error { err_message = Custom _ } -> None) in
 					(match inl with
 					| None -> reduce_expr ctx e

--- a/tests/misc/projects/Issue11180/Main.hx
+++ b/tests/misc/projects/Issue11180/Main.hx
@@ -1,0 +1,15 @@
+class Main {
+	static function main() {
+		trace(Abs.use(fn)(101));
+	}
+
+	inline static function fn(v:Int):Int
+		return v;
+}
+
+@:callable abstract Abs<T>(T->Int) from T->Int {
+	inline public static function use<T>(fn:T->Int):Abs<T> {
+		return fn;
+	}
+}
+

--- a/tests/misc/projects/Issue11180/compile.hxml
+++ b/tests/misc/projects/Issue11180/compile.hxml
@@ -1,0 +1,1 @@
+-main Main

